supabase/migrations/soft_delete_flashcard_sets.sql

```diff
@@ -0,0 +1,59 @@
+-- Add is_deleted column to public.flashcard_sets
+ALTER TABLE
+  public.flashcard_sets
+ADD
+  COLUMN IF NOT EXISTS is_deleted BOOLEAN NOT NULL DEFAULT FALSE;
+
+-- Add deleted_at column to public.flashcard_sets
+ALTER TABLE
+  public.flashcard_sets
+ADD
+  COLUMN IF NOT EXISTS deleted_at timestamp WITH TIME ZONE;
```

Comment:
For pgsql we can use `timestampz` as a shortcut for `timestamp WITH TIME ZONE`.
See https://www.postgresql.org/docs/14/datatype-datetime.html